controllers/flowlogspipeline/flp_common_objects.go

```diff
@@ -742,9 +742,30 @@ func (b *builder) serviceMonitor() *monitoringv1.ServiceMonitor {
 		flpServiceMonitorObject.Spec.Endpoints[0].TLSConfig = &monitoringv1.TLSConfig{
 			SafeTLSConfig: monitoringv1.SafeTLSConfig{
 				ServerName:         serverName,
-				InsecureSkipVerify: true,
+				InsecureSkipVerify: b.desired.Processor.Metrics.Server.TLS.Provided.InsecureSkipVerify,
 			},
 		}
+		if b.desired.Processor.Metrics.Server.TLS.Provided.CaFile != "" {
+			if b.desired.Processor.Metrics.Server.TLS.Provided.Type == flowslatest.RefTypeConfigMap {
+				flpServiceMonitorObject.Spec.Endpoints[0].TLSConfig.SafeTLSConfig.CA = monitoringv1.SecretOrConfigMap{
+					ConfigMap: &corev1.ConfigMapKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: b.desired.Processor.Metrics.Server.TLS.Provided.Name,
+						},
+						Key: b.desired.Processor.Metrics.Server.TLS.Provided.CaFile,
+					},
+				}
+			} else if b.desired.Processor.Metrics.Server.TLS.Provided.Type == flowslatest.RefTypeSecret {
+				flpServiceMonitorObject.Spec.Endpoints[0].TLSConfig.SafeTLSConfig.CA = monitoringv1.SecretOrConfigMap{
+					Secret: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: b.desired.Processor.Metrics.Server.TLS.Provided.Name,
+						},
+						Key: b.desired.Processor.Metrics.Server.TLS.Provided.CaFile,
+					},
+				}
+			}
+		}
```

Comment:
Do we need to provide ConfigMap or Secret if `InsecureSkipVerify` is true ?

I would suggest to create [an helper](https://github.com/netobserv/network-observability-operator/blob/main/pkg/helper/flowcollector.go) to check both `b.desired.Processor.Metrics.Server.TLS.Provided.Type` and `b.desired.Processor.Metrics.Server.TLS.Provided.InsecureSkipVerify`

Reply:
Done, thanks!